packages/terra-framework-docs/src/terra-dev-site/doc/worklist-data-grid/example/DefaultWorklistDataGrid.jsx

```diff
@@ -1,22 +1,98 @@
-import React from 'react';
+import React, { useRef, useState } from 'react';
 import WorklistDataGrid from 'terra-worklist-data-grid';
 import gridDataJSON from './gridData.json';
 
 const DefaultWorklistDataGrid = () => {
+  const rowSelectionModeRef = useRef();
   const rowHeaderIndex = 0;
   const { cols, rows } = gridDataJSON;
+  const [selectedRows, setSelectedRows] = useState([]);
+  const [hasSelectableRows, setHasSelectableRows] = useState(false);
+
+  const determineSelectedRows = (allRowsSelected, userSelectedRow) => {
+    if (!userSelectedRow) {
+      return [];
+    }
+
+    let remainingSelectedRow = [];
+    if (allRowsSelected) {
+      remainingSelectedRow = userSelectedRow;
+    } else if (selectedRows.includes(userSelectedRow[0])) {
+      // Row Deselect so remove this rowId.
+      remainingSelectedRow = selectedRows.filter(e => (e !== userSelectedRow[0]));
+    } else {
+      // Row Selected so add this rowId.
+      remainingSelectedRow = remainingSelectedRow.concat(selectedRows);
+      remainingSelectedRow.push(userSelectedRow[0]);
+    }
+    return remainingSelectedRow;
+  };
```

Comment:
You should not need this method since the consumers are managing the selection state.